tests/measurements/test_counts.py

```diff
@@ -132,6 +135,45 @@ def circuit():
 
         custom_measurement_process(dev, spy)
 
+    def test_batched_counts_dimension(self, mocker):
+        """Test that the counts function outputs counts of the right size with batching"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=2, shots=n_sample)
+        spy = mocker.spy(qml.QubitDevice, "sample")
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX([0.54, 0.65], wires=0)
+            return qml.counts(qml.PauliZ(0)), qml.counts(qml.PauliX(1))
+
+        sample = circuit()
+
+        assert len(sample) == 2
+        assert np.all([sum(s.values()) == n_sample for batch in sample for s in batch])
+
+        custom_measurement_process(dev, spy)
```

Comment:
Is it not supposed to be at the beginning of the test?

Reply:
I believe it's supposed to be at the end, which is what the other tests are doing. It uses actual device values, so it'd have to run after the device and spy are called.